components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -208,20 +221,36 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
   }
 
   createRule() {
```

Comment:
If I am understanding your changes here (L224-268), you removed the common rule creation solely because of the differences of `id` and `type`. However, when a rule is edited it is not possible to change the `id` or `type`, so this additional code should not be necessary. If you uncovered a case to the contrary, there may be some other issue to resolve, perhaps.

Reply:
thanks for pointing this out, a second look-through helped me find the deeper problem!

When I made these changes, `type` and `id` weren't being set in the update request. On second look, I realized that's because `this.ruleForm.value` only contained `{name: "my new rule", conditions: Array(1)}`. Replacing that variable with `this.ruleForm.controls.<field>.value` seemed to do the trick, allowing us to keep the `convertToRule` as it was 🙌